theano/compile/ops.py

```diff
@@ -623,6 +603,7 @@ class Rebroadcast(gof.Op):
     c_code_and_version = {}
 
     check_input = False
+    __props__ = ("axis")
 
     def __init__(self, *axis):
         self.axis = dict(axis)
```

Comment:
In the init, do this:
```
# Sort them to make sure we merge all possible case.
items = sorted(iteritems(self.axis))
self.axis = OrderedDict(items)
```

and use this import `from theano.compat import OrderedDict`